pythonforandroid/recipes/python2legacy/__init__.py

```diff
@@ -0,0 +1,232 @@
+                    'HOSTPGEN={}'.format(hostpgen),
+                    'CROSS_COMPILE_TARGET=yes',
+                    'INSTSONAME=libpython2.7.so',
+                    _env=env)
+
+            if is_darwin():
+                shprint(sh.cp, join(self.get_recipe_dir(), 'patches', '_scproxy.py'),
+                        join('python-install', 'Lib'))
+                shprint(sh.cp, join(self.get_recipe_dir(), 'patches', '_scproxy.py'),
+                        join('python-install', 'lib', 'python2.7'))
+
+            # reduce python
+            for dir_name in ('test', join('json', 'tests'), 'lib-tk',
+                             join('sqlite3', 'test'), join('unittest, test'),
+                             join('lib2to3', 'tests'), join('bsddb', 'tests'),
+                             join('distutils', 'tests'), join('email', 'test'),
+                             'curses'):
+                shprint(sh.rm, '-rf', join('python-install',
+                                           'lib', 'python2.7', dir_name))
+
+            # info('Copying python-install to dist-dependent location')
+            # shprint(sh.cp, '-a', 'python-install', self.ctx.get_python_install_dir())
+
+            # print('Copying hostpython binary to targetpython folder')
+            # shprint(sh.cp, self.ctx.hostpython,
+            #         join(self.ctx.get_python_install_dir(), 'bin', 'python.host'))
+            # self.ctx.hostpython = join(self.ctx.get_python_install_dir(), 'bin', 'python.host')
+
+        # print('python2legacy build done, exiting for debug')
+        # exit(1)
+
+    def create_python_bundle(self, dirn, arch):
+        info("Filling private directory")
+        if not exists(join(dirn, "lib")):
+            info("lib dir does not exist, making")
+            shprint(sh.cp, "-a",
+                    join("python-install", "lib"), dirn)
+        shprint(sh.mkdir, "-p",
+                join(dirn, "include", "python2.7"))
+
+        libpymodules_fn = join("libs", arch.arch, "libpymodules.so")
+        if exists(libpymodules_fn):
+            shprint(sh.mv, libpymodules_fn, dirn)
+        shprint(sh.cp,
+                join('python-install', 'include',
+                     'python2.7', 'pyconfig.h'),
+                join(dirn, 'include', 'python2.7/'))
+
+        info('Removing some unwanted files')
+        shprint(sh.rm, '-f', join(dirn, 'lib', 'libpython2.7.so'))
+        shprint(sh.rm, '-rf', join(dirn, 'lib', 'pkgconfig'))
+
+        libdir = join(dirn, 'lib', 'python2.7')
+        site_packages_dir = join(libdir, 'site-packages')
+        with current_directory(libdir):
+            removes = []
+            for dirname, root, filenames in walk("."):
+                for filename in filenames:
+                    for suffix in EXCLUDE_EXTS:
+                        if filename.endswith(suffix):
+                            removes.append(filename)
+            shprint(sh.rm, '-f', *removes)
```

Comment:
Minor: We could simplify and kill at least one for since `endswith()` can take a list, e.g.:
```python
for dirname, root, filenames in walk("."):
    for filename in filenames:
        if filename.endswith(EXCLUDE_EXTS):
            removes.append(filename)
```

Reply:
I agree this is minor and out of context since this minor debt was already existing, but just moved.
I've created a ticket to track it https://github.com/kivy/python-for-android/issues/1560

Reply:
Fixed in https://github.com/kivy/python-for-android/commit/1a4830b7ddbb6d2f2d0fec0782f76545219b49b9 of this PR